src/toil_scripts/lib/urls.py

```diff
@@ -70,50 +67,18 @@ def s3am_upload_job(job, file_id, file_name, s3_dir, num_cores, s3_key_path=None
     s3am_upload(fpath=fpath, s3_dir=s3_dir, num_cores=num_cores, s3_key_path=s3_key_path)
 
 
-def _download_s3_url(file_path, url, ceph=None):
+def _download_s3_url(file_path, url, s3_key_path=None):
     """
     Downloads from S3 URL via Boto
 
     :param str file_path: Path to file
     :param str url: S3 URL
     """
-    from boto.s3.connection import S3Connection
-    s3 = S3Connection()
-    try:
-        parsed_url = urlparse(url)
-        if not parsed_url.netloc or not parsed_url.path.startswith('/'):
-            raise ValueError("An S3 URL must be of the form s3:/BUCKET/ or "
-                             "s3://BUCKET/KEY. '%s' is not." % url)
-        bucket = s3.get_bucket(parsed_url.netloc)
-        key = bucket.get_key(parsed_url.path[1:])
-        key.get_contents_to_filename(file_path)
-    finally:
-        s3.close()
-
-
-def _download_encrypted_file(url, file_path, key_path):
-    """
-    Downloads encrypted files from S3
-
-    :param str url: URL to be downloaded
-    :param str file_path: Output path to file
-    :param str key_path: Path to the 32-byte key file
-    """
-    # Grab master key
-    with open(key_path, 'r') as f:
-        key = f.read()
-    if len(key) != 32:
-        raise ValueError('Bad key in {}. Must be 32 bytes'.format(key_path))
-
-    key = _generate_unique_key(key_path, url)
-    # Create necessary headers for SSE-C encryption and download
-    encoded_key = base64.b64encode(key)
-    encoded_key_md5 = base64.b64encode(hashlib.md5(key).digest())
-    h1 = 'x-amz-server-side-encryption-customer-algorithm:AES256'
-    h2 = 'x-amz-server-side-encryption-customer-key:{}'.format(encoded_key)
-    h3 = 'x-amz-server-side-encryption-customer-key-md5:{}'.format(encoded_key_md5)
-    subprocess.check_call(['curl', '-fs', '--retry', '5', '-H', h1, '-H', h2, '-H', h3, url, '-o', file_path])
-    assert os.path.exists(file_path)
+    command = ['s3am', 'download']
```

Comment:
Add `--file-exists overwrite` and `--download-exists=discard`. `--download-slots` should also be added, similar to what _s3am_with_retry does. In fact, I would consider refactoring _s3am_with_retry into _s3am_with_retry, _s3am_download_with_retry and _s3am_upload_with_retry by have the latter two call the former.